<robot name="test_robot">

<link name="link1"> 
  <visual>
    <origin xyz="0 0 0.02" />
      <geometry>
        <mesh filename="package://joints_sim/3d/MX-64T.dae" scale="0.001 0.001 0.001"/>
      </geometry>
    <material name="white">
      <color rgba="1 1 1 1"/>
    </material>
  </visual>
</link>

<link name="link2">
  <visual>
    <origin xyz="0 -0.01 0.01" rpy="0 1.57 0"/>
      <geometry>
        <mesh filename="package://joints_sim/3d/RX-64.dae" scale="0.001 0.001 0.001"/>
      </geometry>
    <material name="white">
      <color rgba="1 1 1 1"/>
    </material>
  </visual>
</link>

<link name="link3">
  <visual>
    <origin xyz="0 0 0.04" />
    <geometry>
      <box size=".05 .03 .08" />
    </geometry>
    <material name="white">
      <color rgba="1 1 1 1"/>
    </material>
  </visual>
</link>

<link name="link4">
  <visual>
    <origin xyz="0 0 0.055" rpy="0 3.14 3.14" />
      <geometry>
        <mesh filename="package://joints_sim/3d/face.dae" scale="0.5 0.5 0.5"/>
      </geometry>
    <material name="white">
      <color rgba="1 1 1 1"/>
    </material>
  </visual>
</link>
 
<joint name="joint1" type="continuous">
  <parent link="link1"/>
  <child link="link2"/>
  <origin xyz="0 0.02 0.04" rpy="0 0 0" />
  <axis xyz="0 0 1" />
</joint>
 
<joint name="joint2" type="continuous">
  <parent link="link2"/>
  <child link="link3"/>
  <origin xyz="0 0.02 0.03" rpy="-1.57 0 0" />
  <axis xyz="1 0 0" />
</joint>
 
<joint name="joint3" type="continuous">
  <parent link="link3"/>
  <child link="link4"/>
  <origin xyz="0 0 0.08" rpy="0 0 0" />
  <axis xyz="-1 0 0" />
</joint>

</robot>
